lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -182,7 +197,9 @@ void DeclAndTypeClangFunctionPrinter::printFunctionSignature(
     CFunctionSignatureTypePrinter typePrinter(
         os, cPrologueOS, typeMapping, outputLang, interopContext,
         FunctionSignatureTypeUse::ReturnType);
-    typePrinter.visit(objTy, retKind);
+    // Pass false because it's a parameter for indirect return.
+    // Not an actual parameter, which means it can never be inout
+    typePrinter.visit(objTy, retKind, false);
```

Comment:
NIT: You can add a comment before `false` to explain the param - `/*isInOutParam=*/false`, and then you can shorten the comments you have above it to reduce the justification.